cpp/src/endpoint.cpp

```diff
@@ -61,14 +61,15 @@ Endpoint::Endpoint(std::shared_ptr<Component> workerOrListener,
   params->err_handler.arg = _callbackData.get();
 
   if (worker->isProgressThreadRunning()) {
-    utils::CallbackNotifier callbackNotifier{UCS_INPROGRESS};
+    utils::CallbackNotifier callbackNotifier{false};
     auto worker = ::ucxx::getWorker(_parent);
-    worker->registerGenericPre([this, &params, &callbackNotifier]() {
+    auto status = UCS_INPROGRESS;
```

Comment:
I've also made the initial `status = UCS_INPROGRESS`. This is not something that `ucp_ep_create` will return, but we can use it ensure that it has changed after `ucp_ep_create` is called (we want either `UCS_OK` or `UCS_ERR_*`). In essence, the original code was correct w.r.t. to the `UCS_INPROGRESS` comparison, but the `ucsErrorThrow(status)` was wrong, since it was being cast down to a `bool`.